### PR TITLE
Add WithdrawInbox.sol

### PR DESCRIPTION
**Pull request by ClaudeZsb**

### CODE REVIEW

File: contracts/WithdrawInbox.sol [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     */
+    function withdraw(
+        address _receiver,
+        uint64 _toChain,
+        uint64[] calldata _fromChains,
+        address[] calldata _tokens,
+        uint32[] calldata _ratios,
+        uint32[] calldata _slippages
+    ) external {
+        emit WithdrawalRequest(msg.sender, _receiver, _toChain, _fromChains, _tokens, _ratios, _slippages);
```
Comment (xiaozhou) [inline contracts/WithdrawInbox.sol:37]: add require checks for equal lengths of arrays

File: contracts/WithdrawInbox.sol [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    );
+
+    /**
+     * @notice Withdraw liquidity from the pool-based bridge.
+     * NOTE: ONLY call this from a contract address. DO NOT call from an EOA.
+     * NOTE: _fromChains, _tokens, _ratios, _slippages should have the same length.
+     * @param _receiver The receiver address on _toChain.
+     * @param _toChain The chainId of chain to which withdrew tokens would be transferred.
+     * @param _fromChains THe chainId of chains from which withdrew tokens would be transferred.
+     * @param _tokens The tokens to be withdrew.
```
Comment (xiaozhou) [inline contracts/WithdrawInbox.sol:25]: simplify comments:
```
_toChain The chain Id to receive the withdrawn tokens
_fromChains The chain Ids to withdraw tokens
_tokens The token to withdraw on each fromChain
```

File: contracts/WithdrawInbox.sol [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+     * @param _ratios The withdrawal ratios of each token.
+     * @param _slippages The max slippages of each token for cross-chain transfer.
```
Comment (xiaozhou) [inline contracts/WithdrawInbox.sol:27]: cross-chain withdraw

File: contracts/WithdrawInbox.sol [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        uint64 toChain,
+        uint64[] fromChains,
+        address[] tokens,
+        uint32[] ratios,
+        uint32[] slippages
+    );
+
+    /**
+     * @notice Withdraw liquidity from the pool-based bridge.
+     * NOTE: ONLY call this from a contract address. DO NOT call from an EOA.
```
Comment (Dominator008) [inline contracts/WithdrawInbox.sol:20]: Not needed? This can be called from an EOA

File: contracts/WithdrawInbox.sol [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+     * NOTE: _fromChains, _tokens, _ratios, _slippages should have the same length.
```
Comment (Dominator008) [inline contracts/WithdrawInbox.sol:21]: Add as requires instead of NOTE?

File: contracts/WithdrawInbox.sol [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+     * @param _receiver The receiver address on _toChain.
+     * @param _toChain The chainId of chain to which withdrew tokens would be transferred.
```
Comment (Dominator008) [inline contracts/WithdrawInbox.sol:23]: s/withdrew/withdrawn here and below.

File: contracts/WithdrawInbox.sol [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+     */
+    function withdraw(
+        address _receiver,
+        uint64 _toChain,
+        uint64[] calldata _fromChains,
+        address[] calldata _tokens,
+        uint32[] calldata _ratios,
+        uint32[] calldata _slippages
+    ) external {
+        require(_fromChains.length == _tokens.length, "length mismatch");
```
Comment (Dominator008) [inline contracts/WithdrawInbox.sol:35]: I think a single require is enough and saves bytecode.

```solidity
require (_tokens.length == _fromChains.length && _ratios.length == _fromChains.length && _slippages.length == _fromChains.length, "lengths mismatch");
```

Comment (ClaudeZsb) [reply contracts/WithdrawInbox.sol:35]: ok

**Review by xiaozhou — APPROVED** [review]
`npx prettier -w contracts` before merge

**Review by Jircs1 — COMMENTED** [review]
contracts/WithdrawInbox.sol

**Pull request merged**